### ISSUE TITLE
GEOS-CHEM ERROR running Full-chemistry with GCAP2.0

### ISSUE THREAD

**Issue by Racha711** [geoschem/geos-chem#2107]
### Name and Institution (Required)

Name: Racha
Institution: Kyoto University

### Confirm you have reviewed the following documentation

- [Y ] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/help-and-reference/SUPPORT.html)
- [Y ] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [ Y] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [ Y] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question

GCClassic version 14.1.1
I have run the Full chemistry using GCAP2.0 under SSP3-7.0 in 2090-2099 period and found that it was stopped at 20901201 with the error message as below;

```console
===============================================================================
GEOS-Chem ERROR: KPP failed to converge after 2 iterations!
 -> at Do_FullChem (in module GeosCore/FullChem_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Incorrect species units after Do_FullChem!
 -> at Do_Chemistry  (in module GeosCore/chemistry_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Do_Chemistry"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
     - CLEANUP: deallocating arrays now...

-------------------------------------------------------------------------------
HEMCO 3.6.2 FINISHED.
Warnings (level 1 or lower):  37260
-------------------------------------------------------------------------------
```
Kindly, consider this situation.

Sincerely,
Racha

[GC_ssp370.txt](https://github.com/geoschem/geos-chem/files/13857707/GC_ssp370.txt)

Please provide as much detail as possible. Always include the GEOS-Chem version number and any relevant configuration and log files.

**Comment by yantosca** [comment]
Thanks for writing @Racha711.  It looks like most of the species concentrations have gone to NaN in the problem box:
```console
 #############################################################################################################################################################################################################################################
 ### KPP DEBUG OUTPUT!
 ### Species concentrations at problem box           20           3          16
 #############################################################################################################################################################################################################################################
 ###                        NaN CH2I2
 ###                        NaN CH2IBr
 ###                        NaN CH2ICl
 ###                        NaN AERI
 ###                        NaN AONITA
 ###                        NaN CO2
 ###                        NaN INDIOL
 ###                        NaN ISALA
 ###                        NaN ISALC
 ###                        NaN LBRO2H
 ###                        NaN LBRO2N
 ###                        NaN BRO2
 ###                        NaN LISOPOH
 ###                        NaN LISOPNO3
 ###                        NaN LNRO2H
 ###                        NaN LNRO2N
 ###                        NaN NRO2
 ###                        NaN NAP
 ###                        NaN LTRO2H
 ###                        NaN LTRO2N
 ###                        NaN TRO2
 ###                        NaN LVOCOA
 ###                        NaN LVOC
 ###                        NaN LXRO2H
 ###                        NaN LXRO2N
 ###                        NaN XRO2
 ###                        NaN MSA
 ###                        NaN SO4s
 ###                        NaN SOAGX
 ###                        NaN SOAIE
 ###                        NaN POx
 ###                        NaN LOx
 ###                        NaN PCO
 ###                        NaN LCO
 ###                        NaN PSO4
 ###                        NaN LCH4
 ###                        NaN PH2O2
 ###                        NaN I2O4
 ###                        NaN CH3CCl3
 ###                        NaN I2O2
 ###                        NaN MONITA
 ###                        NaN CFC114
 ###                        NaN CH3I
 ###                        NaN H1301
 ###                        NaN H2402
 ###                        NaN CFC115
 ###                        NaN I2O3
 ###                        NaN CFC113
 ###                        NaN PPN
 ###                        NaN BrNO2
 ###                        NaN CCl4
 ###                        NaN CFC11
 ###                        NaN CFC12
 ###                        NaN C2H2
 ###                        NaN H1211
 ###                        NaN INO
 ###                        NaN N2O
 ###                        NaN NIT
 ###                        NaN NITs
 ###                        NaN BENZ
 ###                        NaN N
 ###                        NaN OCS
 ###                        NaN PAN
 ###                        NaN ETHN
 ###                        NaN HI
 ###                        NaN MAP
 ###                        NaN BZCO3H
 ###                        NaN CHCl3
 ###                        NaN IBr
 ###                        NaN CH2Br2
 ###                        NaN MPN
 ###                        NaN Cl2O2
 ###                        NaN CH2Cl2
 ###                        NaN CHBr3
 ###                        NaN C2H4
 ###                        NaN TOLU
 ###                        NaN XYLE
 ###                        NaN HCFC123
 ###                        NaN HCFC141b
 ###                        NaN HCFC142b
 ###                        NaN HCFC22
 ###                        NaN DMS
 ###                        NaN HMHP
 ###                        NaN HMML
 ###                        NaN HMS
 ###                        NaN IPRNO3
 ###                        NaN MENO3
 ###                        NaN ETNO3
 ###                        NaN NPRNO3
 ###                        NaN OIO
 ###                        NaN R4P
 ###                        NaN RA3P
 ###                        NaN RB3P
 ###                        NaN RP
 ###                        NaN BZPAN
 ###                        NaN IONITA
 ###                        NaN CH3Br
 ###                        NaN ICl
 ###                        NaN BALD
 ###                        NaN CH3Cl
 ###                        NaN HNO4
 ###                        NaN ETO
 ###                        NaN ClOO
 ###                        NaN PYAC
 ###                        NaN BENZP
 ###                        NaN IDC
 ###                        NaN OClO
 ###                        NaN ETP
 ###                        NaN CSL
 ###                        NaN PP
 ###                        NaN PRPN
 ###                        NaN ALK4
 ###                        NaN ETHP
 ###                        NaN IEPOXD
 ###                        NaN MVKDH
 ###                        NaN HPALD2OO
 ###                        NaN PHEN
 ###                        NaN PIP
 ###                        NaN SO4
 ###                        NaN HPALD1OO
 ###                        NaN C3H8
 ###                        NaN IDCHP
 ###                        NaN INA
 ###                        NaN HPALD4
 ###                        NaN Br2
 ###                        NaN HPALD3
 ###                        NaN IEPOXA
 ###                        NaN IEPOXB
 ###                        NaN EOH
 ###                        NaN MCRDH
 ###                        NaN HONIT
 ###                        NaN BrCl
 ###                        NaN MACR1OOH
 ###                        NaN MP
 ###                        NaN SALCAL
 ###                        NaN IHN2
 ###                        NaN IHN3
 ###                        NaN MCT
 ###                        NaN IDHDP
 ###                        NaN I2
 ###                        NaN C4HVP1
 ###                        NaN C4HVP2
 ###                        NaN IDNOO
 ###                        NaN SALAAL
 ###                        NaN AROMP5
 ###                        NaN ICNOO
 ###                        NaN INPD
 ###                        NaN ISOPNOO2
 ###                        NaN MPAN
 ###                        NaN MTPA
 ###                        NaN MTPO
 ###                        NaN MVKPC
 ###                        NaN RIPA
 ###                        NaN ROH
 ###                        NaN AROMP4
 ###                        NaN BENZO
 ###                        NaN C2H6
 ###                        NaN RIPB
 ###                        NaN IDHPE
 ###                        NaN MCRENOL
 ###                        NaN RIPD
 ###                        NaN ISOPNOO1
 ###                        NaN IDHNDOO2
 ###                        NaN IDHNDOO1
 ###                        NaN LIMO
 ###                        NaN MVKHC
 ###                        NaN RIPC
 ###                        NaN HPETHNL
 ###                        NaN N2O5
 ###                        NaN ICHE
 ###                        NaN MCRHNB
 ###                        NaN BrNO3
 ###                        NaN H
 ###                        NaN MONITS
 ###                        NaN ETOO
 ###                        NaN BZCO3
 ###                        NaN INPB
 ###                        NaN AROMRO2
 ###                        NaN IHPOO1
 ###                        NaN IHPOO2
 ###                        NaN MVKHCB
 ###                        NaN HPALD1
 ###                        NaN IHPOO3
 ###                        NaN HPALD2
 ###                        NaN IHPNDOO
 ###                        NaN CH4
 ###                        NaN BENZO2
 ###                        NaN HNO2
 ###                        NaN HC5A
 ###                        NaN ICHOO
 ###                        NaN CH3CHOO
 ###                        NaN Cl2
 ###                        NaN ATOOH
 ###                        NaN PROPNN
 ###                        NaN MCRHN
 ###                        NaN MONITU
 ###                        NaN PRN1
 ###                        NaN R4N2
 ###                        NaN IONO
 ###                        NaN MVKOHOO
 ###                        NaN MCROHOO
 ###                        NaN ICPDH
 ###                        NaN MACR1OO
 ###                        NaN PO2
 ###                        NaN ETHLN
 ###                        NaN NPHEN
 ###                        NaN HCOOH
 ###                        NaN H2O2
 ###                        NaN ITCN
 ###                        NaN IHN4
 ###                        NaN OLNN
 ###                        NaN OLND
 ###                        NaN ETO2
 ###                        NaN MOH
 ###                        NaN ACTA
 ###                        NaN IHN1
 ###                        NaN IHPNBOO
 ###                        NaN GLYX
 ###                        NaN ACET
 ###                        NaN ISOP
 ###                        NaN LIMO2
 ###                        NaN MEK
 ###                        NaN IO
 ###                        NaN IEPOXAOO
 ###                        NaN IEPOXBOO
 ###                        NaN MVKHP
 ###                        NaN MCRHP
 ###                        NaN MGLY
 ###                        NaN IDHNBOO
 ###                        NaN CH2OO
 ###                        NaN ClNO2
 ###                        NaN GLYC
 ###                        NaN A3O2
 ###                        NaN PIO2
 ###                        NaN OTHRO2
 ###                        NaN ICN
 ###                        NaN MVKN
 ###                        NaN ITHN
 ###                        NaN IDN
 ###                        NaN INO2B
 ###                        NaN INO2D
 ###                        NaN IHOO1
 ###                        NaN IHOO4
 ###                        NaN MVK
 ###                        NaN MACRNO2
 ###                        NaN HAC
 ###                        NaN MACR
 ###                        NaN ATO2
 ###                        NaN KO2
 ###                        NaN RCO3
 ###                        NaN R4N1
 ###                        NaN R4O2
 ###                        NaN B3O2
 ###                        NaN PRPE
 ###                        NaN RCHO
 ###                        NaN HOBr
 ###                        NaN MCO3
 ###                        NaN CO
 ###                        NaN ClNO3
 ###                        NaN HNO3
 ###                        NaN ALD2
 ###                        NaN CH2O
 ###                        NaN MO2
 ###                        NaN HOI
 ###                        NaN I
 ###                        NaN IONO2
 ###                        NaN HOCl
 ###                        NaN O1D
 ###                        NaN OH
 ###                        NaN NO2
 ###                        NaN HCl
 ###                        NaN HBr
 ###                        NaN ClO
 ###                        NaN BrO
 ###                        NaN BrSALA
 ###                        NaN BrSALC
 ###                        NaN O3
 ###                        NaN SALACL
 ###                        NaN H2O
 ###                        NaN NO
 ###                        NaN HO2
 ###                        NaN Br
 ###                        NaN SALCCL
 ###                        NaN O
 ###                        NaN Cl
 ###                        NaN SO2
 ###                        NaN NO3
```
I wonder if this could be due to bad or corrupted input meteorology.  Try checking the input data on the day of the failure, especially the meteorology.  (Try to open the data with a netCDF viewer like ncview or Panoply.)

Also there have been a few chemistry fixes since 14.1.1.  You might want to try again with 14.2.3.

You can also try to compile and run with the `-DCMAKE_RELEASE_TYPE=Debug` flag, which will add error-checking for e.g. array out-of-bounds and/or division by zero errors.

**Comment by yantosca** [comment]
Closing out this issue
